tests/Unit/Evolution/DiscontinuousGalerkin/Limiters/Test_HwenoImpl.cpp

```diff
@@ -24,6 +24,7 @@
 #include "Domain/Mesh.hpp"
```

Comment:
The commit message `Update Hweno test` makes it sound like the tests were broken before (at least to me). Is this true and should this get squashed into another commit? If not could you expand a bit in the commit message on why the test needs updating and what parts? There's obviously... umm... a lot of changes in this commit that would be a bit overwhelming to all check to answer this question myself :)

Reply:
Before this commit, the test primarily checked the HWENO logic by testing various calls to `secondary_neighbors_to_exclude_from_fit` and `hweno_modified_neighbor_solution` (which itself calls `secondary_neighbors_to_exclude_from_fit` and `solve_constrained_fit`).

After this commit the check of `hweno_modified_neighbor_solution` is replaced by checks of `solve_constrained_fit` directly (via more detailed testing of the fit itself than before) and the new "high level" function `hweno_impl` which bundles everything together.

The reasons to change which functions are being tested are:
- directly testing `solve_constrained_fit` allows for more granular testing of the fitting procedure.
- `hweno_modified_neighbor_solution` is about to be removed, so before I remove it I want a robust test of the algorithm that is independent of this function.
- `hweno_impl` is the new interface, so needs to be tested.

I know this is quite a lot of change... but I don't think the changes are logically part of either adding the `hweno_impl` function, or part of removing the `hweno_modified_neighbor_solution` function. This is why I kept this a separate "Update the test" commit.

However I'm sure I could clarify the commit message.. do you have any suggestions on wording? e.g., should I paste the entire above description into the commit message?

Reply:
Okay, thanks for explaining that. I'm fine with this being a separate commit, sorry if I gave the impression I wasn't!

How about something like:
```
Update and add more tests for HWENO

- Replace hweno_modified_neighbor_solution with solve_constrained_fit,
  which tests the fit itself in more detail, rather than just the output of the limiter
- Adds a test for the function hweno_impl, which bundles everything together
  and checks that the limiter essentially in its entirety is working correctly
```
for the commit message?

Reply:
Looks good. I'll expand the commit message with that (or slightly edited)

Reply:
Okay, perfect! Thanks :)